common/app/routes/Challenges/Child-Container.jsx

```diff
@@ -1,20 +1,56 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import { challengeSelector } from '../../redux';
+import { challengeUpdated } from './redux';
 
 import CompletionModal from './Completion-Modal.jsx';
 import AppChildContainer from '../../Child-Container.jsx';
+import { OverlayLoader } from '../../helperComponents';
+
+const mapStateToProps = createSelector(
+  challengeSelector,
+  challenge => {
+    const { description } = challenge;
+    return {
+      challenge,
+      showLoading: !description || description.length === 0
+    };
+  }
+);
+
+const mapDispatchToProps = { challengeUpdated };
 
 const propTypes = {
-  children: PropTypes.node
+  challenge: PropTypes.object,
+  challengeUpdated: PropTypes.func.isRequired,
+  children: PropTypes.node,
+  showLoading: PropTypes.bool
 };
 
-export default function ChildContainer({ children, ...props }) {
-  return (
-    <AppChildContainer { ...props }>
-      { children }
-      <CompletionModal />
-    </AppChildContainer>
-  );
+class ChildContainer extends PureComponent {
+  componentDidUpdate(prevProps) {
+    const { challenge = {}, challengeUpdated } = this.props;
+    if (prevProps.showLoading && !this.props.showLoading) {
+      challengeUpdated(challenge);
```

Comment:
Why are you dispatching a challenge updated here?

Reply:
Because `state.file` does not update when new challenge data is available, which means the editor doesn't render with that challenge seed (classic) or `Panes` fails to find the correct component (modern). 

This appropriately describes what is actually happening when we receive new `block` data

Reply:
If the goal is to update the preview plane why not add the action actually related to the event of updating blocks to the updateMain epic so we have a direct cause->effect relationship. 

https://github.com/freecodecamp/freecodecamp/blob/staging/common/app/routes/Challenges/redux/execute-challenge-epic.js#L45

Reply:
It is the challenge seed that is the problem, I need to dispatch an action that creates the Poly with the correct content for the editor.

Reply:
Is the problem here that the action that changes the showLoading from true->false can't trigger the files reducer to also update?